plugins/poll/assets/javascripts/discourse/components/poll-results-ranked-choice.gjs

```diff
@@ -7,15 +7,15 @@ export default class PollResultsRankedChoiceComponent extends Component {
   get rankedChoiceWinnerText() {
     return htmlSafe(
       I18n.t("poll.ranked_choice.winner", {
-        count: this.args.rankedChoiceOutcome.round_activity.length,
-        winner: this.args.rankedChoiceOutcome.winning_candidate.html,
+        count: this.args.rankedChoiceOutcome?.round_activity?.length,
+        winner: this.args.rankedChoiceOutcome?.winning_candidate?.html,
       })
     );
   }
```

Comment:
Hi @nattsw this scenario never practically happens for the user because the tab is not shown until there is a result.

However, during the rendering process, for a short time, the votes can be empty whilst it loads in which causes an exception.  At that point the rendering stops and the results are never shown leading to the screen shot supplied.

Looking at core code is it common practice to use optional chaining operator for similar reasons?

Reply:
Resolved with https://github.com/discourse/discourse/pull/28542/commits/44085796d766bb9d7a0cd49bf953b98d7d2e2572